File: client/src/lib/invoke/contactAdd.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";

export function contactAdd(
    login: string,
    nickname: string,
    notes: string,
    favorite: boolean
): ResultAsync<void, string> {
    return ResultAsync.fromPromise(
        invoke("contact_add", { login, nickname, notes, favorite }),
        (error) => `Failed to add contact: ${error}`
    );
}

File: client/src-tauri/src/commands/contact_add.rs
use crate::core::contacts::{Contact, ContactBook};

#[tauri::command]
pub fn contact_add(
    login: String,
    nickname: String,
    notes: String,
    favorite: bool,
) -> Result<(), String> {
    let mut book = ContactBook::load();
    let contact = Contact {
        login,
        nickname,
        notes,
        favorite,
        created_at: std::time::SystemTime::now()
            .duration_since(std::time::UNIX_EPOCH)
            .map(|d| d.as_secs())
            .unwrap_or(0),
        last_used: 0,
    };
    book.add_contact(contact)
}
